docs/astro/sql/operators/load_file.rst

```diff
@@ -0,0 +1,272 @@
+
+#. **use_native_support** - Native paths support is available for some FileSource and Databases, if it is available the default is to use this path. To leverage these paths certain settings/changes need to be done on destination databases. If for some reason users don't want to use these paths they can turn off this behavior by passing ``use_native_support=False``.
+        This feature is enabled by default, to disable it refer to the below code.
+
+        .. literalinclude:: ../../../../example_dags/example_load_file.py
+           :language: python
+           :start-after: [load_file_example_7_start]
+           :end-before: [load_file_example_7_end]
+
+        To check if the native path will be used for data transfer for a combination of file location and database, refer section :ref:`supported_native_path`
+
+#. **native_support_kwargs** - Since we support multiple databases they may require some parameters to process a file or control error rate etc, those parameters can be passed in ``native_support_kwargs``. These parameters are passed to the destination database.
+
+        Check for valid parameters based on **file location** and **database** combination in section :ref:`supported_native_path`
+
+        .. literalinclude:: ../../../../example_dags/example_load_file.py
+           :language: python
+           :start-after: [load_file_example_8_start]
+           :end-before: [load_file_example_8_end]
+
+
+#. **enable_native_fallback** -  When ``use_native_support`` is True, we try to use the native path, and if this fails we try to use the default path to load data, giving the user a warning. If you want to change this behavior pass ``enable_native_fallback=False``.
+
+        .. literalinclude:: ../../../../example_dags/example_load_file.py
+           :language: python
+           :start-after: [load_file_example_9_start]
+           :end-before: [load_file_example_9_end]
+
+.. _supported_native_path:
+
+Supported Native Paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. list-table::
+   :widths: auto
+
+   * - File Location
+     - Database
+     - native_support_kwargs params
+   * - S3
+     - Bigquery
+     - https://cloud.google.com/bigquery-transfer/docs/s3-transfer#bq
+   * - GCS
+     - Bigquery
+     - https://cloud.google.com/bigquery-transfer/docs/cloud-storage-transfer#bq
+   * - S3
+     - Snowflake
+     -
+   * - GCS
+     - Snowflake
+     -
+
+.. _file_location:
+
+Supported File Location
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Users can load file from all the supported file location that are listed below:
+
+.. literalinclude:: ../../../../src/astro/constants.py
+   :language: python
+   :start-after: [filelocation_start]
+   :end-before: [filelocation_end]
+
+.. _filetype:
+
+Supported File Type
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Users can load the file of all the supported file types that are listed below:
```

Comment:
If someone specifies the file format in File(filetype) parameter, and that contradicts the .suffix of the file name, I'm assuming that the filetype specified wins. Can you confirm and document?

Reply:
Added to docs